src/perfoutputwidgetkonsole.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+  perfoutputwidgetkonsole.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "perfoutputwidgetkonsole.h"
+
+#include <QEvent>
+#include <QKeyEvent>
+#include <QStandardPaths>
+#include <QTemporaryFile>
+#include <QVBoxLayout>
+
+#include <KParts/ReadOnlyPart>
+#include <KParts/kde_terminal_interface.h>
+#include <KService>
+
+PerfOutputWidgetKonsole::PerfOutputWidgetKonsole(QWidget* parent)
+    : PerfOutputWidget(parent)
+{
+    auto layout = new QVBoxLayout(this);
+    setLayout(layout);
+}
+
+PerfOutputWidgetKonsole::~PerfOutputWidgetKonsole() = default;
+
+PerfOutputWidgetKonsole* PerfOutputWidgetKonsole::create(QWidget* parent)
+{
+    auto part = createPart();
+    if (!part) {
+        return nullptr;
+    }
+
+    delete part;
+
+    if (findTail().isEmpty()) {
+        return nullptr;
+    }
+
+    return new PerfOutputWidgetKonsole(parent);
+}
+
+bool PerfOutputWidgetKonsole::eventFilter(QObject* watched, QEvent* event)
+{
+    Q_UNUSED(watched);
+    if (event->type() == QEvent::KeyPress) {
+        auto keyEvent = static_cast<QKeyEvent*>(event);
+
+        // ignore ctrl + c
+        if (keyEvent->key() == Qt::Key_C && keyEvent->modifiers() == Qt::ControlModifier) {
+            return true;
+        }
+
+        if (m_inputEnabled) {
```

Comment:
hm this is going to interact horribly with TUI applications and or things that use readline internally

can't we just eat the events when input is disabled, but otherwise let the konsole handle the events itself ?

Reply:
no, this would be too easy